File: bankingonrails/app/models/bank.rb
class Bank < ApplicationRecord


  composed_of :bIC,
    class_name: "BIC",
    mapping: [
      %w[bIC_value value]
    ]

  has_many :Branches, class_name: 'Branch'
  has_many :Products, class_name: 'BankingProduct'
  has_many :Customers, class_name: 'Customer'
  has_many :Accounts, class_name: 'Account'
  has_many :PaymentCards, class_name: 'PaymentCard'
  has_many :LoanAccounts, class_name: 'LoanAccount'
  has_many :ExchangeRates, class_name: 'ExchangeRate'
  has_many :Consents, class_name: 'Consent'
  has_many :ThirdPartyProviders, class_name: 'ThirdPartyProvider'

end

File: bankingonaspdotnet/Domain/Bank.cs
using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class Bank
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? BankId { get; set; }
    public virtual string? Name { get; set; }
    public virtual string? LegalName { get; set; }
    public virtual BIC? SwiftBic { get; set; }
    public virtual string? HeadquartersCountry { get; set; }
    public virtual string? Website { get; set; }
    public virtual ICollection<Branch> Branches { get; set; } = new List<Branch>();
    public virtual ICollection<BankingProduct> Products { get; set; } = new List<BankingProduct>();
    public virtual ICollection<Customer> Customers { get; set; } = new List<Customer>();
    public virtual ICollection<Account> Accounts { get; set; } = new List<Account>();
    public virtual ICollection<PaymentCard> PaymentCards { get; set; } = new List<PaymentCard>();
    public virtual ICollection<LoanAccount> LoanAccounts { get; set; } = new List<LoanAccount>();
    public virtual ICollection<ExchangeRate> ExchangeRates { get; set; } = new List<ExchangeRate>();
    public virtual ICollection<Consent> Consents { get; set; } = new List<Consent>();
    public virtual ICollection<ThirdPartyProvider> ThirdPartyProviders { get; set; } = new List<ThirdPartyProvider>();

    public static Bank FromRequest(BankRequest request)
    {
        return new Bank
        {
            Id = request.Id,
            Name = request.Name,
            LegalName = request.LegalName,
            SwiftBic = request.SwiftBic,
            HeadquartersCountry = request.HeadquartersCountry,
            Website = request.Website,
        };
    }
}
